### ISSUE TITLE
Thoughts on Cargo Economy Minigame

### ISSUE THREAD

**Issue by N3X15** [d3athrow/vgstation13#15643]
Just writing this shit down while I'm in the sweet spot of exhaustion.

* Cargo shipments are transfers to/from a (virtual) warehouse.
* Warehouse 
  * Comes pre-stocked with some essentials, like medical shit.
  * All warehouse supplies are very finite.
  * To obtain more shit, you have to order them from suppliers like Nanotransen, Discount Dan's, etc.
  * To get rid of shit, you sell it to buyers like the Vox, Nanotransen, etc.
* Suppliers
  * Generated per game but have "specialties" as to what and how much they'll be selling
  * Prices adjusted according to supply.
  * Also have preset prices for stuff you want to sell them based on their "specialties".  Nanotransen will buy fuckloads of plasma for a decent price, for instance, whereas Discount Dan will not pay you *dick* for *anything*.

Warehouse would be capable of storing tens of thousands of crates.  Shuttle can only handle 1 per tile.

tl;dr in order to get shit you have to sell shit.

**Comment by Kurfursten** [comment]
This seems like it might result in Cargo items ending up 'locked' harder than R&D, since you need to even potentially wait for minerals to come in anyway, then wait for sales, send away stuff, put in order, wait for it to arrive, and then delivery.

I could see that upsetting virologists / botanists / assistants who would like to do a gimmick like building or barbering rather than greytide / anyone who likes to put in orders near shift start; read: the period in which you wouldn't have any cargo available.

**Comment by MadmanMartian** [comment]
How about stocks and investments?

Investing in a company may yield certain crates that can be bought after certain gates, and increase the cost of stock in that company whilst also reducing the investment level of other companies so it plays out like a balancing game.

You could even take out sponsorship deals, like installing a number of vending machines around the station where when purchased from it sends the funds 50/50 to the sponsoring company, and to whom set up the deal.

Then, if you've taken stock in a company and the stocks are high, you cash in.

Could even tie this into the newscasters, every interval something happens to one of the companies that may help or hinder their competitive edge that is then reported on by a stock market channel, like the mining company's managed to strike a deal with a rogue mining station so they've got cheaper stock for that interval, or a cargo transport has crashed due to 'pilot incompetence' so that company's lost stock. Basically a degree of randomness and constant change out of the players control that can either be beneficial, or detrimental

**Comment by FudgePucker** [comment]
>Just writing this shit down while I'm in the sweet spot of exhaustion.

Oh yeee I know that feel.
But I agree with kurf in that having it be too difficult to order nessisary supplies every round would be bad.
What about something like goon's where you get a static list of order able crates, and a few other dealers that sell and buy different items that change as the shift goes along?

**Comment by Durgen** [comment]
Unfortunately if you are going to do any sort of balance pass over the economy you are going to piss off miners and they'll mob your PR with mean emojis.

While you are on the subject of cargo, It would be nice to get some sort of trader crates too, as has been suggested before. Giving traders something to spend money on would incentivize trading and just maybe turn it into a real job and not assistant++.
https://github.com/vgstation-coders/vgstation13/issues/13364

**Comment by Blithering** [comment]
Traders are in a bit of a tough spot balance wise, they have a ton of unique and powerful stuff, but very little onus to actually bother trading.

**Comment by Kurfursten** [comment]
When you think about it, few jobs have any incentives to do their job, besides miners.

**Comment by Blithering** [comment]
Most jobs tend to have an element of progression, working towards getting better stuff. Vox have all their stuff round start, unless they do food gimmicks or go vault hunting they won't progress to having better stuff to trade
